src/cmakeTaskProvider.ts

```diff
@@ -39,23 +66,30 @@ export class CMakeTaskProvider implements vscode.TaskProvider {
         }
     }
 
-    public updateDefaultTargets(defaultTargets: string[] | undefined) {
+    public updateDefaultTargets(defaultTargets?: string[]) {
         this.defaultTargets = (defaultTargets && defaultTargets.length > 0) ? defaultTargets :
             this.cmakeDriver ? [this.cmakeDriver.allTargetName] : [CMakeTaskProvider.allTargetName];
     }
 
     public async provideTasks(): Promise<CMakeTask[]> {
         // Create a CMake build task
         const result: CMakeTask[] = [];
-        const taskName: string = "CMake: build";
+        this.updateDefaultTargets();
+        // Provide build task.
+        const taskName: string = localizeCommandType(CommandType.build);
         const definition: CMakeTaskDefinition = {
             type: CMakeTaskProvider.CMakeScriptType,
-            label: taskName,
-            command: "build"
+            label: CMakeTaskProvider.CMakeSourceStr + ": " + taskName,
+            command: CommandType.build,
+            targets: this.defaultTargets
         };
-        const task = new vscode.Task(definition, vscode.TaskScope.Workspace, taskName, CMakeTaskProvider.CMakeSourceStr);
+        const task = new vscode.Task(definition, vscode.TaskScope.Workspace, taskName, CMakeTaskProvider.CMakeSourceStr,
```

Comment:
is `taskName` here supposed to be `definition.label`?  Or do they rebuild the label with `taskName` and `CMakeSourceStr`?